lib/AST/DiagnosticConsumer.cpp

```diff
@@ -29,6 +29,21 @@ using namespace swift;
 
 DiagnosticConsumer::~DiagnosticConsumer() = default;
 
+DiagnosticInfo::FixIt::FixIt(CharSourceRange R, StringRef Str,
+                             ArrayRef<DiagnosticArgument> Args): Range(R) {
+  if (!Args.empty()) {
+    // FIXME: Defer text formatting to later in the pipeline.
```

Comment:
This is definitely going to have ownership questions, but the current implementation is fine.

Reply:
yeah, eventually I think it makes sense for the FixIt to have an ArrayRef<DiagnosticArgument>, and follow the ownership conventions used for error diag arguments. Right now that's hard to do without introducing a circular dependency between DiagnosticConsumer.h and DiagnosticEngine.h, so I decided to tackle that later.